sale_delivery_date_by_zone/models/partner_delivery_zone.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Ooops404
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class PartnerDeliveryZone(models.Model):
+    _name = "partner.delivery.zone"
+    _inherit = ["partner.delivery.zone", "time.cutoff.mixin"]
+
+    order_delivery_cutoff_preference = fields.Selection(
+        [
+            ("warehouse_cutoff", "Use global (warehouse) cutoff time"),
+            ("partner_cutoff", "Use partner's cutoff time"),
+        ],
+        string="Delivery orders cutoff preference",
+        default="warehouse_cutoff",
+        help="Define the cutoff time for delivery orders:\n\n"
+        "* Use global (warehouse) cutoff time: Delivery order for sale orders"
+        " will be postponed to the next warehouse cutoff time\n"
+        "* Use partner's cutoff time: Delivery order for sale orders"
+        " will be postponed to the next partner's cutoff time\n\n"
+        "Example: If cutoff is set to 09:00, any sale order confirmed before "
+        "09:00 will have its delivery order postponed to 09:00, and any sale "
+        "order confirmed after 09:00 will have its delivery order postponed "
+        "to 09:00 on the following day.",
+    )
+
+    delivery_time_preference = fields.Selection(
```

Comment:
This is to define the partner opening hours.
What's the purpose on a delivery zone?

Reply:
This is the default to apply on partners when the delivery zone is selected for them.

In the use case of our customer, it's rare that partners have specific time needs, and they're much more constrained by the courier for that specific zone; and it's tiresome and time consuming to have to fill the info manually on each partner when it's almost always the same within the same delivery zone.

Reply:
Do you use that to schedule the work in the warehouse? Because this was the purpose of defining those values.

Can you elaborate how this is further used in your use case?

Why don't you set on the partner a new selection on order_delivery_cutoff_preference like partner_zone_cutoff and on delivery_time_preference like delivery_zone_window? Those would then read the values from the delivery zone instead of the partner.